src/models/installation.ts

```diff
@@ -127,11 +147,26 @@ Installation.init({
 		type: DataTypes.INTEGER,
 		allowNull: true
 	}
-}, { sequelize });
+}, {
+	hooks: {
+		beforeSave: async (app: Installation, opts) => {
+			await app.encryptChangedSecretFields(opts.fields);
+			if (!app["safeSharedSecret"]) throw new Error("Fail saving app as safeSharedSecret is empty");
```

Comment:
I can't change `allowNull: true` to `allowNull: false` for this new column. I suspect it will be find, but worry sequelize might do something weird on query/select, and fail if this column is empty. So in the definition, set `allowNull: true`, but throw error when save.

Reply:
I don't think you need it for the time being. Plus, last thing we need is for things to break just because the column isn't set and you can't save anything.

Reply:
however, we should add something to set the `encryptedSharedSecret` field when the `sharedSecret` field changes.

Reply:
done.